src/main/jbake/content/documentation/development/osgi-mock.md

```diff
@@ -231,3 +235,192 @@ More examples:
+    public class MyService {
+
+        // specify runtime retention to allow for direct usage in unit tests 
+        @Retention(RetentionPolicy.RUNTIME)
+        public @interface Config {
+            String path() default "/";
+        }
+
+        private final String path;
+
+        @Activate
+        public MyService(Config config) {
+            this.path = config.path();
+        }
+
+        public String getPath() {
+            return path;
+        }
+    }
+
+
+A companion unit test in JUnit 4 might look like this:
+
+    #!java
+    import org.apache.sling.testing.mock.osgi.config.annotations.ApplyConfig;
+    import org.apache.sling.testing.mock.osgi.config.annotations.UpdateConfig;
+    import org.apache.sling.testing.mock.osgi.junit.ConfigCollector;
+    import org.apache.sling.testing.mock.osgi.junit.OsgiContext;
+    import org.apache.sling.testing.mock.osgi.junit.OsgiContextBuilder;
+    import org.junit.Rule;
+
+    import static org.junit.Assert.assertEquals;
+
+    public class MyServiceTest {
+
+        @Rule
+        public OsgiContext context = new OsgiContextBuilder().build();
+
+        @Rule
+        public ConfigCollector configs = new ConfigCollector(context, MyService.Config.class);
+
+        @Test
+        @MyService.Config(path = "/apps") // requires @Retention(RetentionPolicy.RUNTIME)
+        public void myServiceMethod() {
+            MyService.Config config = configs.configStream(MyService.Config.class)
+                    .findFirst()
+                    .orElseThrow();
+            MyService myService = new MyService(config);
+            assertEquals("/apps", myService.getPath());
+        }
+
+        @Test
+        @ApplyConfig(type = MyService.Config.class, property = "path=/libs")
+        public void myServiceMethod() {
+            MyService.Config config = configs.configStream(MyService.Config.class)
+                    .findFirst()
+                    .orElseThrow();
+            MyService myService = new MyService(config);
+            assertEquals("/libs", myService.getPath());
+        }
+
+        @Test
+        @UpdateConfig(pid = "new-pid", property = "path=/content")
+        @ApplyConfig(pid = "new-pid", type = MyService.Config.class)
+        public void myServiceMethod() {
+            MyService.Config config = configs.configStream(MyService.Config.class)
+                    .findFirst()
+                    .orElseThrow();
+            MyService myService = new MyService(config);
+            assertEquals("/content", myService.getPath());
+        }
+    }
+
+### Config Annotations: SlingContext Compatibility
+
+The OSGi Mock Config Annotations and JUnit4/JUnit5 extensions are compatible with the `SlingContext` from Sling Mocks and other libraries that provide extensions of `OsgiContextImpl`.
+
+To setup a project with Sling Mocks and OSGi Mocks Config Annotations, adjust the project pom.xml for either JUnit4 or JUnit5:
+
+For JUnit 5:
+
+    #!xml
+    <dependency>
```

Comment:
I had checked `sling-mock.junit5` for a dependency on `osgi-mock.junit5`, but [it doesn't have one](https://github.com/apache/sling-org-apache-sling-testing-sling-mock/blob/852f247e0c6ef7ed92f8e2684ea1d15c41e1a672/junit5/pom.xml#L37-L49). I just assumed that `sling-mock.junit4` was the same situation, but upon checking it now, I see that [it does have an osgi-mock.junit4 dependency](https://github.com/apache/sling-org-apache-sling-testing-sling-mock/blob/852f247e0c6ef7ed92f8e2684ea1d15c41e1a672/junit4/pom.xml#L51-L55). 

So really, no additional maven configuration should be necessary for use of Config Annotations with either `sling-mock.junit4` or `sling-mock.junit5`, but we'll need to update the `sling-mock.junit5` pom.xml so that it follows this expectation. 

It looks like the same situation with [aem-mock.junit4](https://github.com/wcm-io/io.wcm.testing.aem-mock/blob/8606c9d8b98cd844ba79b31b2fc56b4706b898e0/junit4/pom.xml#L61-L85) and [aem-mock.junit5](https://github.com/wcm-io/io.wcm.testing.aem-mock/blob/8606c9d8b98cd844ba79b31b2fc56b4706b898e0/junit5/pom.xml#L68), so a similar change should be applied to `aem-mock.junit5`'s pom.xml.

Reply:
ah, you're right. each *-mock.junit4/5 module comes with their own specific code, and the dependency chain goes only over the *-mock.core modules which does not contain the actual junit4/5 code.

however, it's still not recommended to include multiple *-mock.junit4/5 modules of osgi/sling/aem in the same project (also it works fine technically), as you usually need only one of them with the fitting (Osgi|Sling|Aem)Context implementation of your choice.

maybe it would make sense to put the new annotations in a separate module for junit5 and 4 in osgi-mock, which then is included as dependency in the other ones. but let's finish the basic discussion in https://github.com/apache/sling-org-apache-sling-testing-osgi-mock/pull/31 first.